solcjs-get

```diff
@@ -25,51 +54,83 @@ function getVersionList (cb) {
   https.get('https://solc-bin.ethereum.org/bin/' + version, function (response) {
     if (response.statusCode !== 200) {
       console.log('Error downloading file: ' + response.statusCode);
       process.exit(1);
     }
+
+    fs.ensureDirSync(compilerDir);
+
+    var file = fs.createWriteStream(targetPath);
+
     response.pipe(file);
     file.on('finish', function () {
       file.close(function () {
-        var hash = '0x' + keccak256(fs.readFileSync(outputName, { encoding: 'binary' }));
-        if (expectedHash !== hash) {
-          console.log('Hash mismatch: ' + expectedHash + ' vs ' + hash);
-          process.exit(1);
-        }
         console.log('Done.');
+        if (cb) cb(targetPath);
       });
     });
   });
 }
 
-console.log('Downloading correct solidity binary...');
-
-getVersionList(function (list) {
-  list = JSON.parse(list);
-  var wanted = pkg.version.match(/^(\d+\.\d+\.\d+)$/)[1];
-  var releaseFileName = list.releases[wanted];
-  var expectedFile = list.builds.filter(function (entry) { return entry.path === releaseFileName; })[0];
-  if (!expectedFile) {
-    console.log('Version list is invalid or corrupted?');
-    process.exit(1);
-  }
-  var expectedHash = expectedFile.keccak256;
-  downloadBinary('soljson.js', releaseFileName, expectedHash);
-});
+if (argv.list) {
+  console.log('Getting the list of all versions ...');
+  getVersionList(function (list) {
+    list = JSON.parse(list).builds;
+    for (var i = list.length - 1; i >= 0; i--) {
+      console.log(list[i].version, list[i].path);
+    }
+    process.exit(0);
+  });
+} else if (argv.clean) {
+  console.log('Removing all local compilers in ' + compilerDir + ' ...');
+  fs.remove(compilerDir, function (err) {
+    if (err) return console.error(err);
+
+    console.log('Success! Cleaned ' + compilerDir);
+    process.exit(0);
+  });
+} else if (argv.all) {
+  console.log('Getting all the versions ...');
+  getVersionList(function (list) {
+    list = JSON.parse(list).builds;
+    for (var i = list.length - 1; i >= 0; i--) {
+      var target = list[i].path;
+      if (target.indexOf('nightly') > 0 && !argv.nightly) continue;
+
+      downloadBinary(target);
+    }
+  });
+} else if (argv.releases) {
+  console.log('Getting all the releases ...');
+  getVersionList(function (list) {
+    list = JSON.parse(list).releases;
+    for (var key in list) {
+      downloadBinary(list[key]);
+    }
+  });
+} else {
+  getVersionList(function (list) {
+    list = JSON.parse(list);
+
+    var wanted = null;
+    if (requestedVersion) {
+      console.log('Requested version: ' + requestedVersion);
+      wanted = requestedVersion;
+    } else {
+      console.log('Requested version: latest release');
+      wanted = list.releases[pkg.version.match(/^(\d+\.\d+\.\d+)$/)[1]];
+    }
+
+    downloadBinary(wanted, function (file) {
+      if (!requestedVersion){
+        fs.copy(file, 'soljson.js'); // for backward compatibility
+      }
```

Comment:
This should really be a symlink (at least if the FS supports it).

Actually, this does not seem like something just for backwards compatibility. Can solc-js even use the compiler when it's inside `bin/`? I think that an option to relink it to another version might be useful because that's basically how the select the compiler version now.